src/main/java/org/truffleruby/parser/MethodTranslator.java

```diff
@@ -496,14 +496,18 @@ public RubyNode visitZSuperNode(ZSuperParseNode node) {
                 this,
                 argsNode);
 
-        final SequenceNode reloadSequence = (SequenceNode) reloadTranslator.visitArgsNode(argsNode);
+        final RubyNode reloadSequence = reloadTranslator.visitArgsNode(argsNode);
```

Comment:
I think it'd be a lot cleaner if `ReloadArgumentsTranslator#visitArgsNode` returns a RubyNode[], because a SequenceNode is conceptually wrong there, since we need the values of all nodes, not just the last one. I'll do that (will need to not use visitArgsNode but another name, not a problem).